store/tikv/coprocessor.go

```diff
@@ -173,7 +173,12 @@ func (r *copRanges) slice(from, to int) *copRanges {
 	if to <= len(r.mid) {
 		ran.mid = r.mid[from:to]
 	} else {
-		ran.mid, ran.last = r.mid[from:], r.last
+		if from <= len(r.mid) {
+			ran.mid = r.mid[from:]
+		}
+		if from < to {
```

Comment:
Then it will be incorrect. `from < to` is specially for the case `from == to`, when the result should be empty slice.